templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -345,52 +306,140 @@ contents:
       done
 
       echo "WARN: OVS did not succesfully activate NM connection. Attempting to bring up connections"
       counter=0
       while [ $counter -lt 5 ]; do
-        if nmcli conn up ovs-if-br-ex; then
+        if nmcli conn up "$ovs_interface"; then
           echo "OVS successfully configured"
+          update_nm_conn_files "$bridge_name" "$port_name"
           copy_nm_conn_files
-          ip a show br-ex
+          ip a show "$bridge_name"
           ip route show
           nmcli c show
           configure_driver_options ${iface}
-          exit 0
+          configured=true
+          break
         fi
         sleep 5
         counter=$((counter+1))
       done
+      if [ $configured != true ]; then
+        echo "ERROR: Failed to activate $ovs_interface NM connection"
+        exit 1
+      fi
+    }
 
-      echo "ERROR: Failed to activate ovs-if-br-ex NM connection"
+    if ! rpm -qa | grep -q openvswitch; then
+      echo "Warning: Openvswitch package is not installed!"
       exit 1
+    fi
+
+    echo "Current routing and connection state:"
+    ip route show
+    nmcli c show
+
+    if [ "$1" == "OVNKubernetes" ]; then
+      # Configures NICs onto OVS bridge "br-ex"
+      # Configuration is either auto-detected or provided through a config file written already in Network Manager
+      # key files under /etc/NetworkManager/system-connections/
+      # Managing key files is outside of the scope of this script
+
+      # if the interface is of type vmxnet3 add multicast capability for that driver
+      # REMOVEME: Once BZ:1854355 is fixed, this needs to get removed.
+      extra_bridge_file='/etc/ovnk/extra_bridge'
+      function configure_driver_options {
+        intf=$1
+        if [ ! -f "/sys/class/net/${intf}/device/uevent" ]; then
+          echo "Device file doesn't exist, skipping setting multicast mode"
+        else
+          driver=$(cat "/sys/class/net/${intf}/device/uevent" | grep DRIVER | awk -F "=" '{print $2}')
+          echo "Driver name is" $driver
+          if [ "$driver" = "vmxnet3" ]; then
+            ifconfig "$intf" allmulti
+          fi
+        fi
+      }
+      iface=""
+      counter=0
+      # find default interface
+      while [ $counter -lt 12 ]; do
+        # check ipv4
+        iface=$(ip route show default | awk '{ if ($4 == "dev") { print $5; exit } }')
+        if [[ -n "$iface" ]]; then
+          echo "IPv4 Default gateway interface found: ${iface}"
+          break
+        fi
+        # check ipv6
+        iface=$(ip -6 route show default | awk '{ if ($4 == "dev") { print $5; exit } }')
+        if [[ -n "$iface" ]]; then
+          echo "IPv6 Default gateway interface found: ${iface}"
+          break
+        fi
+        counter=$((counter+1))
+        echo "No default route found on attempt: ${counter}"
+        sleep 5
+      done
+
+      if [ "$iface" != "br-ex" ] && [  -f "$extra_bridge_file" ] && nmcli connection show br-ex &> /dev/null; then
```

Comment:
This is to rollback the temporary solution currently used in some deployments.
They use https://github.com/trozet/openshift-ovn-migration to create a br-ex out of an interface different from the main one. The idea here is that if we are in that scenario (br-ex present and not the default gateway, plus the presence of the file), then we roll back

Reply:
Actually, I can leave a comment on top of that

Reply:
I see. Now I don't understand how we were not breaking things for them before where we did things if `br-ex` was not the main interface?

Reply:
Because they are using it in local gw mode, so all the traffic goes to the kernel and eventually finds the right route

Reply:
But I mean if `br-ex` exists but is not their main interface, when the current version of configure-ovs runs for them it will attempt to create a main br-ex interface? So I don't understand why we workaround their situation in this PR, but we did not in the current version.

Reply:
Ah, now I get it, and I am not 100% sure. 
Maybe (but I don't know the full details) every time the node respins it recreates br-ex and the other script runs after this one moving br-ex out of the gw interface. I see the script running after this one: https://github.com/trozet/openshift-ovn-migration/blob/main/ovs-migration.service#L8

Reply:
I guess if that is the case, there's no need for rolling back. @trozet WDYT?

Reply:
Post migration, the script will not return because the default gateway iface will not be br-ex. However, the script has checks to not change any of the nm conns if they already exist:
https://github.com/openshift/machine-config-operator/blob/release-4.8/templates/common/_base/files/configure-ovs-network.yaml#L200

The only thing the script may do (and possibly cause failure) is:
https://github.com/openshift/machine-config-operator/blob/release-4.8/templates/common/_base/files/configure-ovs-network.yaml#L181

The migration script works around it by ensuring the device is brought back up. We should probably fix configure-ovs to not bring the device down using device disconnect, and not bring down anything if the nm file is already there and conn is up.

When the customer deploys 4.9 with shared gateway, they wont have the migration service enabled via ignition anymore. Therefore this script needs to handle the case where we need to rollback and rewire br-ex correctly to the default gateway interface. I don't see the harm in handling that here.

@jcaamano ideally the migration should have been included as part of MCO (and this script), but due to time constraints it is added via ignition separately.